### PR TITLE
Bug 1419670, Update backup and restore procedure

### PR DESCRIPTION
**Pull request by ahardin-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=1419670

Preview builds:
http://file.rdu.redhat.com/~ahardin/03172017/etcd-backup-restore/admin_guide/backup_restore.html

http://file.rdu.redhat.com/~ahardin/03172017/etcd-backup-restore/install_config/downgrade.html

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@sttts This is my first pass at updating the docs for the issues described in the BZ. Can you please take a look?

Also, you mention that the following is an incorrect location. What should this be updated to?

 etcdctl -C https://${ETCD_CA_HOST}:2379 \
  --ca-file=/etc/etcd/ca.crt     \
  --cert-file=/etc/etcd/peer.crt     \
  --key-file=/etc/etcd/peer.key member add ${NEW_ETCD} https://${NEW_ETCD_IP}:2380

Thank you!

File: admin_guide/backup_restore.adoc [hunk 1]
```diff
@@ -320,6 +320,15 @@ events. If the backup is still larger than 700 MB, stop the other hosts before
 performing this step.
 ====
 +
+[WARNING]
```
Comment (sttts) [inline admin_guide/backup_restore.adoc]: Why not format the next step in a way that it is part of this? In the online version this appear as 5b, the next as 6. It should be all 5b or 5b + 5c.

**Comment by ahardin-rh** [comment]
@sttts Thanks! This is updated.

Preview build: http://file.rdu.redhat.com/~ahardin/03022017/etcd-backup-restore/admin_guide/backup_restore.html

**Comment by ahardin-rh** [comment]
@adellape @bmcelvee @mburke5678 please peer review

**Comment by mburke5678** [comment]
@ahardin-rh 
The only quibble I have is over the word "complain."
_"Otherwise,etcd will also **complain** that the `cluster-id` does not match"_
 Maybe use "report" or "indicate"??

**Comment by sttts** [comment]
Can we merge 5b and 5c? This is still confusing. 5c does not make sense without 5b. Then I would drop the sentence "If you do not apply this step, the following step will break the restore."

**Comment by ahardin-rh** [comment]
@mburke5678 @sttts Thanks. This is updated.

File: admin_guide/backup_restore.adoc (hunk 1 above)
Comment (sttts) [inline admin_guide/backup_restore.adoc]: Looks much better now 👍

Comment (sttts) [inline admin_guide/backup_restore.adoc]: Side note: can you update the TTL text above? I doubt this has anything to do with voltage levels in this case. "Time to live" is a better translation I guess ;-)

**Comment by sttts** [comment]
lgtm

File: admin_guide/backup_restore.adoc (hunk 1 above)
Comment (ahardin-rh) [inline admin_guide/backup_restore.adoc]: @sttts Fixed. Thanks! :smile:

**Comment by ahardin-rh** [comment]
@adellape @bfallonf I had to make some changes after discussion with QE. Can you please give this a sanity check/peer review? Thanks! :mag_right:

File: admin_guide/backup_restore.adoc [hunk 2]
```diff
@@ -177,6 +176,276 @@ that {product-title} was previously installed.
 # chown -R etcd:etcd $ETCD_DATA_DIR
 ----
 
+. Create the new single node cluster using etcd's `--force-new-cluster`
+option. You can do this with a long complex command using the values from
+*_/etc/etcd/etcd.conf_*, or you can temporarily modify the *systemd* unit file
+and start the service normally.
++
+To do so, edit the *_/usr/lib/systemd/system/etcd.service_* and add
```
Comment (bfallonf) [inline admin_guide/backup_restore.adoc]: "edit the X file, and add"

or just

"edit X, and add"

File: admin_guide/backup_restore.adoc [hunk 3]
```diff
@@ -177,6 +176,276 @@ that {product-title} was previously installed.
...
++
+[WARNING]
+====
+The `$ETCD_DIR` location differs between external and embedded etcd.
+====
+
+. Restore your *_/etc/etcd/etcd.conf_* file from backup or *_.rpmsave_*.
+
+. Create the new single node cluster using etcd's `--force-new-cluster`
+option. You can do this with a long complex command using the values from
```
Comment (bfallonf) [inline admin_guide/backup_restore.adoc]: I'd suggest the "long complex" bit be taken out. Adverbs!

"Do this by using the values from..."

File: admin_guide/backup_restore.adoc [hunk 4]
```diff
@@ -177,6 +176,276 @@ that {product-title} was previously installed.
...
+    --key /etc/etcd/peer.key \
+    https://172.18.1.18:2379/v2/members/511b7fb6cc0001 \
+    -XPUT -H "Content-Type: application/json" \
+    -d '{"peerURLs":["https://172.18.1.18:2380"]}'
+----
+
+. Re-run the `member list` command and ensure the peer URLs no longer include
+*localhost*.
+
+. Now add each additional member to the cluster, one at a time.
```
Comment (bfallonf) [inline admin_guide/backup_restore.adoc]: Should this link somewhere?

**Comment by bfallonf** [comment]
@ahardin-rh Just a couple comments from me. I think there's a larger conversation around the "embedded etcd? vs the "etcd on the master host", but this might not be the place. There's a bit of confusion around it on sme list. I'll bring it up in the team meeting tomorrow.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../admin_guide/backup_restore.adoc#admin-guide-backup-and-restore[Backup and Restore]
|Updated the backup and restore procedure.
%

**Pull request merged**